### DISCUSSION TITLE
Question: How to avoid "MolsToGridImage" to show Hydrogen atoms

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4032] [Q&A]
I am using "MolsToGridImage" to draw following smiles. 

```
smile_1 = 'C(C)N'
smile_2 = 'O(C(C)C)C(C)(C)O'
smiles = [smile_1, smile_2]
mols = [Chem.MolFromSmiles(smile) for smile in smiles]
legends = smiles
Draw.MolsToGridImage(mols, molsPerRow=len(mols), subImgSize=(200, 200), legends=legends)
```

The resulting image is: 

<img width="389" alt="With_Hydrogen" src="https://user-images.githubusercontent.com/38423900/114654609-84a9bf80-9d25-11eb-8852-c98c5ff78c1f.png">

However, I would like to get: 

<img width="389" alt="Without_Hydrogen" src="https://user-images.githubusercontent.com/38423900/114654645-92f7db80-9d25-11eb-955c-401251457797.png">

I am looking for a way to avoid showing hydrogen atoms. If you have any idea, please let me know.

**Comment by jasondbiggs** [comment]
One method would be to explicitly set the label for every non-carbon atom to its atomic symbol.  Something like

```
def setLabels(mol):
    for atom in mol.GetAtoms():
        if atom.GetAtomicNum() != 6:
            atom.SetProp("_displayLabel",atom.GetSymbol())
```
then using your `mols` you can do

```
for mol in mols:
    setLabels(mol)
legends = smiles
Draw.MolsToGridImage(mols, molsPerRow=len(mols), subImgSize=(200, 200), legends=legends) 
```
<img width="361" alt="image" src="https://user-images.githubusercontent.com/16337306/114734999-6387a580-9d0a-11eb-90c5-faa58c0a1d65.png">

**Reply by mohammad-saber** [reply]
Thank you. It works.

**Comment by mohammad-saber** [comment]
I am trying to add a sign like * to the atom that hydrogen was removed from it (similar to sample image below). 

<img width="398" alt="Screen Shot 2021-04-19 at 13 36 55" src="https://user-images.githubusercontent.com/38423900/115182511-08dcb800-a115-11eb-837c-8219749baa2a.png">



If you have any idea about it, I really appreciate it to share with me. Thank you.
